app/src/main/java/org/mozilla/fenix/perf/ProfilerUtils.kt

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.provider.DocumentsContract
+import android.util.Base64
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import org.json.JSONObject
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import java.io.BufferedReader
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.InputStreamReader
+import java.util.zip.GZIPInputStream
+
+private val profiler_api = "https://api.profiler.firefox.com/compressed-store"
+private val profiler_header = "application/vnd.firefox-profiler+json;version=1.0"
+private val token = "profileToken"
+private val profileDataURL = "https://profiler.firefox.com/public/"
+
+private val firefox_features = arrayOf("screenshots", "js", "leaf", "stackwalk", "cpu", "java", "processcpu", "ipcmessages", "java")
```

Comment:
question: will these get out-of-date with the settings that the `about:debugging` UI uses? I wonder if there's a way we can use those values to avoid duplication. Maybe in a follow-up.